src/listeners/command-handler/CoreMessageParser.ts

```diff
@@ -16,16 +16,24 @@ export class CoreListener extends Listener<typeof Events.PreMessageParsed> {
 		if (!canRun) return;
 
 		let prefix = null;
-		const mentionPrefix = this.getMentionPrefix(message.content);
+		const userMentionPrefix = this.getUserMentionPrefix(message.content);
+		const managedRoleMentionPrefix = this.getManagedRoleMentionPrefix(message);
```

Comment:
Just do this, so the role mention prefix logic isn't needlessly run when the user mention prefix matched:
```suggestion
		const mentionPrefix = this.getUserMentionPrefix(message.content) ??this.getManagedRoleMentionPrefix(message);
```

And revert the code below, it's duplicated.